File: src/LMS_Assignment.Domain/Entities/StudentEnrollment.cs
using LMS_Assignment.Domain.Enums;

namespace LMS_Assignment.Domain.Entities;

public class StudentEnrollment
{
    public Guid Id { get; set; }
    public Guid StudentId { get; set; }
    public Guid ClassId { get; set; }
    public string? RollNumber { get; set; }
    public EnrollmentStatus Status { get; set; } = EnrollmentStatus.Active;
    public DateTime EnrolledAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public User Student { get; set; } = null!;
    public Class Class { get; set; } = null!;
}

File: frontend/src/types/student-enrollment.ts
export interface StudentEnrollment {
  id: string
  studentId: string
  studentName: string
  classId: string
  className: string
  rollNumber: string | null
  status: string
  enrolledAt: string
}

export interface CreateStudentEnrollmentRequest {
  studentId: string
  classId: string
  rollNumber?: string
}
